### ISSUE TITLE
GitAhead theme does not apply correctly anymore

### ISSUE THREAD

**Issue by Gr3q** [gitahead/gitahead#443]
On Arch Linux, with dark theme:

![image](https://user-images.githubusercontent.com/22377521/84487421-a1551100-ac96-11ea-9852-600b10dad1f5.png)

**Comment by hackhaslam** [comment]
Can you tell which version of GitAhead broke?

**Comment by Gr3q** [comment]
@hackhaslam 2.6.2

**Comment by hackhaslam** [comment]
Version 2.6.2 doesn't use Qt 5.15.0. Are you using the official binaries or building your own?

**Comment by Gr3q** [comment]
building my own, because the prebuilt one does not work correctly with Arch Linux (using keychains).

I'll rebuild it now, just to see if it helps

**Comment by sphaerophoria** [comment]
Fixed the QPainterPath issues and PR'd #444.

Can confirm that self built version against Qt 5.15.0 has the same issue

**Comment by sphaerophoria** [comment]
I see the following with a light theme
![image](https://user-images.githubusercontent.com/1069811/84557207-f0726480-acdd-11ea-8853-f5af319294bb.png)
if I make the following change the commit list at least seems to get a little better
```
diff --git a/src/ui/CommitList.cpp b/src/ui/CommitList.cpp
index a744e55..46dee28 100644
--- a/src/ui/CommitList.cpp
+++ b/src/ui/CommitList.cpp
@@ -676,9 +676,9 @@ public:
     auto textRole = selected ? QPalette::HighlightedText : QPalette::Text;
     auto brightRole = selected ? QPalette::WindowText : QPalette::BrightText;
     QPalette palette = Application::theme()->commitList();
     QColor text = palette.color(group, textRole);
-    QColor bright = palette.color(group, brightRole);
+    QColor bright = Qt::black;

     painter->save();
     painter->setRenderHints(QPainter::Antialiasing);
```
![image](https://user-images.githubusercontent.com/1069811/84557229-1b5cb880-acde-11ea-9846-da10632b4189.png)

Looking at the commitList palette it looks like we were defaulting to the default value of "BrightText" in the palette
```
QPalette Theme::commitList()
{
  QPalette palette;
  QColor bright = palette.color(QPalette::BrightText);
```

It's possible that this has changed in a newer version of Qt

**Comment by sphaerophoria** [comment]
Default application palette behaving differently seems very likely. Looking at the diff in qtbase from 5.14.2 and 5.15.0 

It went from
```
static void initSystemPalette()
{
    if (QApplicationPrivate::sys_pal)
        return; // Already initialized

    QPalette defaultPalette;
    if (QApplicationPrivate::app_style)
        defaultPalette = QApplicationPrivate::app_style->standardPalette();

    auto *platformTheme = QGuiApplicationPrivate::platformTheme();
    if (const QPalette *themePalette = platformTheme ? platformTheme->palette() : nullptr) {
        QApplicationPrivate::setSystemPalette(themePalette->resolve(defaultPalette));
        QApplicationPrivate::initializeWidgetPaletteHash();
    } else {
        QApplicationPrivate::setSystemPalette(defaultPalette);
    }
}
```
to
```
void QApplicationPrivate::initializeWidgetPalettesFromTheme()
{
    QPlatformTheme *platformTheme = QGuiApplicationPrivate::platformTheme();
    if (!platformTheme)
        return;

    widgetPalettes.clear();

    struct ThemedWidget { const char *className; QPlatformTheme::Palette palette; };

    static const ThemedWidget themedWidgets[] = {
        { "QToolButton", QPlatformTheme::ToolButtonPalette },
        { "QAbstractButton", QPlatformTheme::ButtonPalette },
        { "QCheckBox", QPlatformTheme::CheckBoxPalette },
        { "QRadioButton", QPlatformTheme::RadioButtonPalette },
        { "QHeaderView", QPlatformTheme::HeaderPalette },
        { "QAbstractItemView", QPlatformTheme::ItemViewPalette },
        { "QMessageBoxLabel", QPlatformTheme::MessageBoxLabelPalette },
        { "QTabBar", QPlatformTheme::TabBarPalette },
        { "QLabel", QPlatformTheme::LabelPalette },
        { "QGroupBox", QPlatformTheme::GroupBoxPalette },
        { "QMenu", QPlatformTheme::MenuPalette },
        { "QMenuBar", QPlatformTheme::MenuBarPalette },
        { "QTextEdit", QPlatformTheme::TextEditPalette },
        { "QTextControl", QPlatformTheme::TextEditPalette },
        { "QLineEdit", QPlatformTheme::TextLineEditPalette },
    };

    for (const auto themedWidget : themedWidgets) {
        if (auto *palette = platformTheme->palette(themedWidget.palette))
            QApplication::setPalette(*palette, themedWidget.className);
    }
}
```
Note the lack of a defaultPalette case in the new version.

**Comment by sphaerophoria** [comment]
Nice that looks like it was it. The following diff seems to have solved it
```
diff --git a/src/app/Application.cpp b/src/app/Application.cpp
index 7a54b89..22b5177 100644
--- a/src/app/Application.cpp
+++ b/src/app/Application.cpp
@@ -26,8 +26,9 @@
 #include <QNetworkProxyFactory>
 #include <QNetworkReply>
 #include <QOperatingSystemVersion>
 #include <QSettings>
+#include <QStyle>
 #include <QSysInfo>
 #include <QTimer>
 #include <QTranslator>
 #include <QUrlQuery>
@@ -186,8 +187,11 @@ Application::Application(int &argc, char **argv, bool haltOnParseError)
 
   // Enable system proxy auto-detection.
   QNetworkProxyFactory::setUseSystemConfiguration(true);
 
+  // Set default pallete. Pre 5.15.0 this should be the same as the default
+  QApplication::setPalette(QApplication::style()->standardPalette());
+
   // Do platform-specific initialization.
 #if defined(Q_OS_MAC)
   // Register service on Mac.
   registerService();

```

I'll prep a PR once I do a little more testing

![image](https://user-images.githubusercontent.com/1069811/84558157-88278100-ace5-11ea-9573-d026db3f3cce.png)

**Comment by hackhaslam** [comment]
It may have to be ifdefed Linux only. The note at https://doc.qt.io/qt-5/qstyle.html#standardPalette says:

> Note that on systems that support system colors, the style's standard palette is not used. In particular, the Windows Vista and Mac styles do not use the standard palette, but make use of native theme engines. With these styles, you should not set the palette with QApplication::setPalette().

I don't know if that means that you it breaks the Mac and Windows styles. It also doesn't say anything about when that function was introduced.

I will start testing 5.15 very soon and I can test this change on all three platforms.

**Comment by sphaerophoria** [comment]
Ah that's a good point. I don't have a windows/mac environment to test on atm, so I apologize if I have to leave that up to you.

**Comment by hackhaslam** [comment]
I think this is an unintended regression in 5.15. It isn't easy to work around on macOS with the QMacStyle. I've submitted a bug report here: https://bugreports.qt.io/browse/QTBUG-85188.

**Comment by hackhaslam** [comment]
This was fixed in Qt 5.15.1.
